shared/src/main/java/uk/gov/di/authentication/shared/helpers/CookieHelper.java

```diff
@@ -118,6 +118,15 @@ public static Optional<String> parsePersistentCookie(Map<String, String> headers
         }
         final String persistentId = cookieValues[0];
 
+        // Temporary measure, as commit 75a10df4376397d5a454b87b5cee689e13a71e20 introduced a bug
+        // whereby persistent session ID could end up as --<timestamp>--<timestamp>. In these cases
+        // we should just generate a new one to get back on track. It will be possible to remove
+        // this 13 months after it is first merged(== the expiry time of that cookie at the point of
+        // issue in November 2023)
+        if (persistentId.startsWith("--")) {
+            return Optional.empty();
+        }
+
```

Comment:
Since the ID generator just returns random b64url encoded bytes, I think it's probably possible for the ID part of this to start with `--`. Perhaps we ought to add a check to the ID generator to prevent that?

Reply:
That's very true. Does the new proposed code handle this, where instead of checking `startsWith` we apply a fuller regex pattern match?